TCAT/Controllers/AllStopsTableViewController.swift

```diff
@@ -117,39 +107,116 @@ class AllStopsTableViewController: UITableViewController {
             // Adding "#" to keys for bus stops that start with a number
             sortedKeys.append("#")
         }
+    }
 
-        return sortedKeys
+    @objc private func backAction() {
+        navigationController?.popViewController(animated: true)
     }
 
-    func setUpTableOnRetry() {
-        // Retry getting data from user defaults
-        self.allStops = SearchTableViewManager.shared.getAllStops()
-        // Set up table information
-        self.sectionIndexes = self.sectionIndexesForBusStop()
-        self.sortedKeys = self.sortedKeysForBusStops()
+    private func setUpLoadingIndicator() {
+        loadingIndicator = LoadingIndicator()
+        if let loadingIndicator = loadingIndicator {
+            view.addSubview(loadingIndicator)
+            loadingIndicator.snp.makeConstraints { make in
+                make.center.equalToSuperview()
+                make.width.height.equalTo(40)
+            }
+        }
+    }
 
-        self.tableView.reloadData()
+    private func getAllStops() -> Future<Response<[Place]>> {
+        return networking(Endpoint.getAllStops()).decode()
     }
 
-    // MARK: - Table view data source
+    /* Get all bus stops and store in userDefaults */
+    private func refreshAllStops() {
+        setUpLoadingIndicator()
+        if let allBusStops = userDefaults.value(forKey: Constants.UserDefaults.allBusStops) as? Data,
+            var busStopArray = try? decoder.decode([Place].self, from: allBusStops) {
+            // Check if empty so that an empty array isn't returned
+            if !busStopArray.isEmpty {
+                // TODO: Move to backend
```

Comment:
can we make a task for this or post in #transit-backend about this

Reply:
https://github.com/cuappdev/ithaca-transit-backend/issues/242